pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -671,9 +671,58 @@ func getServerByName(client *gophercloud.ServiceClient, name types.NodeName) (*S
 	return &serverList[0], nil
 }
 
-func nodeAddresses(srv *servers.Server, networkingOpts NetworkingOpts) ([]v1.NodeAddress, error) {
+// IP addresses order:
+// * interfaces private IPs
+// * access IPs
+// * metadata hostname
+// * server object Addresses
+func nodeAddresses(srv *servers.Server, interfaces []attachinterfaces.Interface, networkingOpts NetworkingOpts) ([]v1.NodeAddress, error) {
 	addrs := []v1.NodeAddress{}
 
+	// parse private IP addresses first in an ordered manner
+	for _, iface := range interfaces {
+		for _, fixedIP := range iface.FixedIPs {
+			isIPv6 := net.ParseIP(fixedIP.IPAddress).To4() == nil
+			if !(isIPv6 && networkingOpts.IPv6SupportDisabled) {
```

Comment:
We don't need to check IPv6 here and L707, `getAddressByName` has a argument `needIPv6` for that.

Reply:
Though `getAddressByName` func has this option, the `nodeAddresses` func can be called from other places.